### ISSUE TITLE
New lightkurve API can't bin folded, user generated data

### ISSUE THREAD

**Issue by christinahedges** [lightkurve/lightkurve#927]
I'm trying to fold and bin a user generated light curve

The following code produces an error message that is hard to debug
```python
lk.LightCurve(np.arange(10), np.ones(10)).fold(1).bin(3)
```
produces
```python
ValueError: Input values did not match the format class jd:
TypeError: for jd class, input should be doubles, string, or Decimal, and second values are only allowed for doubles.
```
#### Environment

-  platform (e.g. Linux, OSX, Windows): OSX
-  lightkurve version (e.g. 1.0b6): 2.0b4
-  installation method (e.g. pip, conda, source): source

**Comment by barentsen** [comment]
Thanks so much for reporting @christinahedges!  This looks like a real bug.

I am able to reproduce the crash, though I see a different error message:
```python
>>> lk.LightCurve(np.arange(10), np.ones(10)).fold(1).bin(3)
IndexError: index -1 is out of bounds for axis 0 with size 0
```

It looks like the error is triggered by the behavior of this edge case in AstroPy:
```python
>>> from astropy.time import Time
>>> from astropy import units as u
>>> from astropy.timeseries import aggregate_downsample, TimeSeries
>>> ts = TimeSeries(time=Time([0], format='jd'))
>>> aggregate_downsample(ts, time_bin_size=1*u.day)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-84-44ec4dc9a4a0> in <module>
      4 
      5 ts = TimeSeries(time=Time([0], format='jd'))
----> 6 aggregate_downsample(ts, time_bin_size=1*u.day)

~/Library/Caches/pypoetry/virtualenvs/lightkurve-HqDeGDfc-py3.8/lib/python3.8/site-packages/astropy/timeseries/downsample.py in aggregate_downsample(time_series, time_bin_size, time_bin_start, n_bins, aggregate_func)
    100 
    101     # Create new binned time series
--> 102     binned = BinnedTimeSeries(time_bin_start=bins[:-1], time_bin_end=bins[-1])
    103 
    104     # Determine rows where values are defined

~/Library/Caches/pypoetry/virtualenvs/lightkurve-HqDeGDfc-py3.8/lib/python3.8/site-packages/astropy/timeseries/binned.py in __init__(self, data, time_bin_start, time_bin_end, time_bin_size, n_bins, **kwargs)
    160                     times = time_bin_start.copy()
    161                     times[:-1] = times[1:]
--> 162                     times[-1] = time_bin_end
    163                     time_bin_end = times
    164                 time_bin_size = (time_bin_end - time_bin_start).sec * u.s

~/Library/Caches/pypoetry/virtualenvs/lightkurve-HqDeGDfc-py3.8/lib/python3.8/site-packages/astropy/time/core.py in __setitem__(self, item, value)
   1178         if self.scale is not None:
   1179             value = getattr(value, self.scale)
-> 1180         self._time.jd1[item] = value._time.jd1
   1181         self._time.jd2[item] = value._time.jd2
   1182 

IndexError: index -1 is out of bounds for axis 0 with size 0
```

**Comment by nksaunders** [comment]
To add to this, I encountered the same error as @christinahedges trying to bin a folded TESS SPOC light curve, so I don't think it's limited to user-generated data.

**Comment by christinahedges** [comment]
This is still an issue, it's not possible to bin a folded light curve.

**Comment by barentsen** [comment]
This should be fixed now via #953. Thanks @jorgemarpa!!
